Updating changes to Approver and Reviewer (#591)
kubeflow/community: wgs/community-membership.md

Reviewer: I adjusted it a bit to clarify that this is not for root approver / leads, but owners of subprojects, becasue we have many owner files, usually one per subproject.

> an [OWNERS] file (for repos using the bot).

Reviewer: ```suggestion
a non-root [OWNERS] file (for repositories using the bot).
```

- member for at least 3 months

> - Reviewer for or author of at least 5 substantial PRs to the codebase

Reviewer: ```suggestion
- Reviewer for or author of at least 5 substantial PRs to the relevant codebase
```

an [OWNERS] file (for repos using the bot).

- member for at least 3 months

> - Primary reviewer for at least 3 PRs to the codebase

Reviewer 2: 3 **substantial** PRs?

Author: @terrytangyuan do you have an alternative ?  I would like to update and finish this update.

Author: ok, I changed in "Reviewer for or author of at least 5 substantial PRs to the codebase" in 2nd commit

> - Reviewed or merged at least 5 substantial PRs to the codebase

Reviewer 2: I would clarify what substantial means:

Perhaps add something like "with the definition of substantial subject to the lead's discretion (e.g. refactors, enhancements rather than grammar correction or one-line pulls)" similar to what we are doing at Argo project.

Reviewer 3: +1. As in the [Argo doc](https://github.com/argoproj/argoproj/blob/master/community/membership.md), we can also combine these 2 points to one `Reviewer for or author of at least 5 substantial PRs to the codebase,`

Reviewer 4: +1 to Johnu's wording.

Author: ok, I changed to "Reviewer for or author of at least 5 substantial PRs to the codebase" in 2nd comment

> in an [OWNERS] file (for repos using the bot).

Reviewer: ```suggestion
in a non-root [OWNERS] file (for repositories using the bot).
```

in an [OWNERS] file (for repos using the bot).

- Reviewer of the codebase for at least 3 months
- Primary reviewer for at least 5 substantial PRs to the codebase

> - Reviewed or merged at least 10 PRs to the codebase

Reviewer 2: This only mentioned reviewing PRs. I think we want to promote people who actually authored substantial PRs.

Reviewer 3: +1. Similar comment as before. Combining to one, `Reviewer for or author of at least 10 substantial PRs to the codebase`

Reviewer 4: +1 to Johnu's wording.

Author: ok, I updated to "Reviewer for or author of at least 5 substantial PRs to the codebase" in 2nd commit
